save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestService.kt

```diff
@@ -74,16 +74,22 @@ class TestService(
     fun getTestBatches(agentId: String): Mono<TestBatch> {
         val agent = agentRepository.findByContainerId(agentId) ?: error("The specified agent does not exist")
         log.debug("Agent found, id=${agent.id}")
-        val execution = agent.execution
-        val lock = locks.computeIfAbsent(execution.id!!) { Any() }
+        val executionId = agent.execution.id!!
+        val lock = locks.computeIfAbsent(executionId) { Any() }
         return synchronized(lock) {
+            log.debug("Acquired lock for executionId=$executionId")
             val testExecutions = transactionTemplate.execute {
+                val execution = executionRepository.getById(executionId)
```

Comment:
Turns out, execution must be read inside the same transaction, where it will be modified.

Reply:
extremely strange. 70% sure that is not the reason